### PR TITLE
feat: remove MinOut safety from THOR repays

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Ensures we benefit from automagical streaming loan closes, so that users don't end up with reverts and get the best rate possible on their collateral refund.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A, spotted in https://discord.com/channels/554694662431178782/876953436908822568/1244772064590561280

Low, we now honor the THOR-returned memo. If something is wrong here, then something is wrong in the quote.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Test with either your own loan or a frame injected known borrower
- Get a quote for 100% repay and note the memo in the `close` request (see screenshots)
- Ensure that the actual memo in the calldata (either on the on-chain Tx, or in frame if testing with Frame) is the same as the quote one, except it has affiliate name and fees slapped to it

### Engineering

- ^

### Operations

- ^

## Screenshots (if applicable)

- Quote and `close` request

Screenshot 2024-05-28 at 09 46 59

- Actual memo: `$-:ETH.ETH:0x0e1799c427da1881a72912840d40495769b94f6f::ss:0`

Screenshot 2024-05-28 at 09 47 30

### CODE REVIEW

File: src/pages/Lending/Pool/components/Repay/RepayInput.tsx [hunk 1]
```diff
@@ -383,11 +383,11 @@ export const RepayInput = ({
...
     if (!confirmedQuote) return null
 
-    // No need for slippage deduction here - quoteWithdrawnAmountAfterFeesThorBaseUnit (expected_amount_out) already is quote.fees.slippage_bps deducted
-    const minCollateralOut = confirmedQuote.quoteWithdrawnAmountAfterFeesThorBaseUnit
-    return addLimitToMemo({ memo: confirmedQuote.quoteMemo, limit: minCollateralOut })
+    return assertAndProcessMemo(confirmedQuote.quoteMemo)
   }, [confirmedQuote])
 
+  debugger
+
```
Comment (gomesalexandre) [inline src/pages/Lending/Pool/components/Repay/RepayInput.tsx:390]: rm

**Comment by NeOMakinG** [comment]
## Testing
Response from the thor node:
image

Tenderly simulation: 
image

Memo seems sane, the limit has been removed, affiliate link is here

I can't test the full process as I would need a loan to repay but I managed to try it as much as possible with `0x0E1799C427dA1881A72912840D40495769b94f6f`

**Review by NeOMakinG — APPROVED** [review]
Looks good to me, manual tests looks sane and nothing to say on the code side! 👍

**Review by 0xApotheosis — APPROVED** [review]
Sane indeed, source debugs showing memo is processed correctly without the limit, but with the affiliate code added:

Screenshot 2024-05-28 at 8 51 35 PM

Screenshot 2024-05-28 at 8 48 34 PM

**Pull request merged**